src/aof.c

```diff
@@ -1963,37 +1963,64 @@ static int rioWriteHashIteratorCursor(rio *r, hashTypeIterator *hi, int what) {
 /* Emit the commands needed to rebuild a hash object.
  * The function returns 0 on error, 1 on success. */
 int rewriteHashObject(rio *r, robj *key, robj *o) {
+    int res = 0; /*fail*/
+
     hashTypeIterator *hi;
     long long count = 0, items = hashTypeLength(o, 0);
 
+    /* Is expected also hash-fields with expiration (HFE) ? */
```

Comment:
The comment may not be necessary. Otherwise, please reword it, a bit hard to understand.